kedro-datasets/pyproject.toml

```diff
@@ -194,16 +194,14 @@ test = [
     "networkx~=2.4",
     "opencv-python~=4.5.5.64",
     "openpyxl>=3.0.3, <4.0",
-    "pandas-gbq>=0.12.0, <0.18.0; python_version < '3.11'",
-    "pandas-gbq>=0.18.0; python_version >= '3.11'",
-    "pandas~=1.3",  # 1.3 for read_xml/to_xml
+    "pandas-gbq>=0.12.0",
+    "pandas~=2.0",
     "Pillow~=9.0",
     "plotly>=4.8.0, <6.0",
     "polars[xlsx2csv, deltalake]~=0.18.0",
     "pre-commit>=2.9.2",
-    "pyarrow>=1.0; python_version < '3.11'",
-    "pyarrow>=7.0; python_version >= '3.11'",  # Adding to avoid numpy build errors
-    "pyodbc~=4.0.35",
+    "pyarrow>=10.0.1",
```

Comment:
`pandas.FeatherDataset` doc tests were failing because with `pandas 2.x` they needed `pyarrow` to be higher than `10.0.1` but let me try to revert this pin, I think it should get the latest version anyway.